### DISCUSSION TITLE
C++ version question, about " for(char c : str) ".

### DISCUSSION THREAD

**Discussion by iitsmel** [microsoft/vscode-cpptools#7152] [Q&A] [answered]
### What happened:
After I typed in this code below, there'a an error. Which I think it is not supposed to appear.
[Other reference I found online.](https://stackoverflow.com/questions/9438209/for-every-character-in-string)

### Code
for (char c : str) {
    dosomethingto(c);
}

### Error
expected a ';' // I think it means ':' supposed to be ';' 

### Environment
- Visual Studio Code Version: 1.53.2

- ( After typed in " g++ --version " in terminal. )
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/4.2.1
Apple clang version 12.0.0 (clang-1200.0.32.29)
Target: x86_64-apple-darwin20.3.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin

- ( After typed in " clang --version " in terminal. )
Apple clang version 12.0.0 (clang-1200.0.32.29)
Target: x86_64-apple-darwin20.3.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin

**Comment by sean-mcmanus** [comment, marked as answer]
This is supposed to work (and works for me), but you need to make sure your cppStandard is set to c++11 or newer. Run C/C++: Log Diagnostics to see what is being used. Also, the "4.2.1" appears to reference a g++ header for an old compiler that doesn't support C++11.

**Reply by iitsmel** [reply]
I saw what's going on! My C++ standard version is 98. After I switched to 11, problem fixed! Thank you!
